src/dll/edbgEcmdDll.C

```diff
@@ -1696,11 +1702,269 @@ uint32_t dllQueryHostMemInfoRanges( const std::vector<ecmdChipTarget> & i_target
 #endif // ECMD_REMOVE_MEMORY_FUNCTIONS
 
 #ifndef ECMD_REMOVE_INIT_FUNCTIONS
+
+#ifdef EDBG_ISTEP_CTRL_FUNCTIONS
+
+static edbgIPLTable g_edbgIPLTable;
+
+/**
+  * @brief Set the IPL mode
+  *
+  * @param enum ipl_mode i_mode - IPL mode (Interactive or Continous IPL mode)
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t setIplMode(enum IplMode_t i_mode)
+{
+    uint32_t rc = ECMD_SUCCESS;
+    
+    /* If IPL mode is set to interactive then, SBE will be set in ISTEP mode
+     * and execute all SBE isteps in chip op mode */
+    if (i_mode == IPL_MODE_ISTEP)
```

Comment:
This is a workarround code and should be rtemoved once sbe_config_update code is set.
Design flow should be , if the user issued istep poweron , edbg should set istep_mofde attribute in devoce tree na d sbe config update istep will take care the scratch pad register update.

Reply:
yes. I have added that comment down in #1734
//This will be actually set by sbe_config_update istep procedure.
//For testing purpose, we are setting it up.